### PR TITLE
Edit and expand configuration file

### PR DESCRIPTION
**Pull request by electronwill**
This edit expands the configuration file to make all the changes
discussed in working sessions with Kerry Aaron Ilan and Will.

### CODE REVIEW

**Comment by electronwill** [comment]
@asmeurer  This one's ready to merge.

**Comment by asmeurer** [comment]
The code blocks should be marked as yaml (`.. code-block:: yaml`). This will make the comments color correctly (and potentially more if the yaml syntax highlighter in Sphinx ever gets smarter).

**Comment by asmeurer** [comment]
Please use code formatting (\`\`code\`\`) where appropriate. Any command should use that instead of single quotes. The default values should perhaps use that as well, to highlight them.

File: docs/source/sample-condarc.rst [hunk 1]
```diff
@@ -0,0 +1,24 @@
+===================
+Sample condarc file
```
Comment (asmeurer) [inline docs/source/sample-condarc.rst]: "Sample .condarc file"

**Comment by asmeurer** [comment]
It's not possible to have a nested table of contents on the left, but we can get around this by putting a table of contents at the top of the file. There is some way to make Sphinx generate one automatically (I think the `toc` directive, but I'm not entirely sure).

**Comment by electronwill** [comment]
@asmeurer OK, should be ready to merge.

**Comment by electronwill** [comment]
@asmeurer Found an even better way to do it, so made one last edit. Should be ready to merge now.

**Pull request merged**